File: yak-gl-util/src/main/java/net/yakclient/opengl/util/Texture.java
package net.yakclient.opengl.util;

public interface Texture {
    boolean hasAlpha();

    void bind();

    int getImageHeight();

    int getImageWidth();

    float getHeight();

    float getWidth();

    int getTextureHeight();

    int getTextureWidth();

    void release();

    TexAggregation generateTexs(VerticeAggregation vertices);

}

File: util/src/main/kotlin/net/yakclient/graphics/util/Texture.kt
package net.yakclient.graphics.util

public interface Texture {
    public val parent: Texture?

    public val height: Int
    public val width: Int
    public val offsetX: Int
    public val offsetY: Int

    public fun subTexture(x: Int, y: Int, width: Int, height: Int): Texture

    public fun bind()
    public fun release()
}